src/appleseed/renderer/modeling/project/project.cpp

```diff
@@ -436,6 +435,17 @@ void Project::on_frame_end(
     impl->m_rendering_timer.measure();
 }
 
+TextureStore& Project::get_texture_store(const ParamArray& params) const
+{
+    if (!impl->m_texture_store)
+    {
+        assert(impl->m_scene.get());
+        impl->m_texture_store.reset(new TextureStore(*impl->m_scene, params));
```

Comment:
Something looks wrong here (tell me if I'm wrong). 

The texture store is cached in `m_texture_store`. But to create a texture_store, you need to give it some parameters. What if `get_texture_store` is called with different parameters ? 

@dictoon What do you think ?

Reply:
This is no longer relevant.